sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/NonAzureOpenAIKeyCredential.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Token credential class for accepting non-Azure OpenAI credential string.
+ */
+public class NonAzureOpenAIKeyCredential implements TokenCredential {
```

Comment:
This should not implement TokenCredential.

Reply:
We are hoping to be able to seamlessly supply this type of credential in the builder method `credentials(...)`. If that is frowned upon, what would be the alternative? I think we were trying to not add custom methods to the `builder`

Reply:
Remove the extension so no longer an issue.